src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> leftNode = last;
+        final Node<T> newNode = new Node<>(leftNode, value, null);
+        last = newNode;
+        if (leftNode == null) {
+            first = newNode;
+        } else {
+            leftNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            final Node<T> left = node(index).prev;
+            final Node<T> newNode = new Node<>(left, value, node(index));
+            node(index).prev = newNode;
```

Comment:
```suggestion
            final Node<T> node = node(index);
            final Node<T> left = node.prev;
            final Node<T> newNode = new Node<>(left, value, node);
            node.prev = newNode;
```
your code invokes method node(index) 3 times! and iterates through whole list 3 times